src/java/org/apache/cassandra/cql3/functions/FormatFcts.java

```diff
@@ -0,0 +1,434 @@
+    {
+        return decimalFormat.format(value);
+    }
+
+    /**
+     * Converts numeric value in a column to a size value of specified unit.
+     * <p>
+     * If the function call contains just one argument - value to convert - then it will be
+     * looked at as the value is of unit 'B' and it will be converted to a value of a unit which is closest to it. E.g.
+     * If a value is (1024 * 1024 + 1) then the unit will be in MiB and converted value will be 1.
+     * <p>
+     * If the function call contains two arguments - value to convert and a unit - then it will be looked at
+     * as the unit of such value is 'B' and it will be converted into the value of the second (unit) argument.
+     * <p>
+     * If the function call contains three arguments - value to covert and source and target unit - then the value
+     * will be considered of a unit of the second argument, and it will be converted
+     * into a value of the third (unit) argument.
+     * <p>
+     * Examples:
+     * <pre>
+     * format_bytes(val) = format_bytes(val, 'B', 'MiB')
+     * format_bytes(val, 'B', 'MiB')
+     * format_bytes(val, 'B', 'GiB')
+     * format_bytes(val, 'KiB', 'GiB')
+     * format_bytes(val, 'MiB') = format_bytes(val, 'B', 'MiB')
+     * format_bytes(val, 'GiB') = format_bytes(val, 'B', 'GiB')
+     * </pre>
+     * <p>
+     * It is possible to convert values of a bigger unit to values of a smaller unit, e.g. this is possible:
+     *
+     * <pre>
+     * format_bytes(val, 'GiB', 'B')
+     * </pre>
+     * <p>
+     * Values can be max of Long.MAX_VALUE, If the conversion produces overflown value, Long.MAX_VALUE will be returned.
+     * Note that the actual return value will be 9223372036854776000 due to the limitations of double precision.
+     * <p>
+     * Supported units are: B, KiB, MiB, GiB
+     * <p>
+     * Supported column types on which this function is possible to be applied:
+     * <pre>INT, TINYINT, SMALLINT, BIGINT, VARINT, ASCII, TEXT</pre>
+     * For ASCII and TEXT types, text of such column has to be a non-negative number.
+     * <p>
+     *
+     * The conversion of negative values is not supported.
+     */
+    public static class FormatBytesFct extends NativeScalarFunction
+    {
+        private static final String FUNCTION_NAME = "format_bytes";
+
+        private FormatBytesFct(AbstractType<?>... argsTypes)
+        {
+            super(FUNCTION_NAME, UTF8Type.instance, argsTypes);
+        }
+
+        @Override
+        public ByteBuffer execute(Arguments arguments) throws InvalidRequestException
+        {
+            if (arguments.get(0) == null)
+                return null;
+
+            if (arguments.containsNulls())
+                throw new InvalidRequestException("none of the arguments may be null");
+
+            long value = (long) getValue(arguments);
+
+            if (value < 0)
+                throw new InvalidRequestException("value must be non-negative");
```

Comment:
This validation logic seems to be duplicated. Could we extract it to a private method?